### PR TITLE
Add search results page

### PR DESCRIPTION
**Pull request by LizBaker**
You can test this out by going to this path `/search-results/?query=node%20agent&page=1`

### CODE REVIEW

File: src/pages/search-results.js [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+    (async () => {
+      const defaultSources = locale.isDefault
+        ? ['developer', 'docs', 'opensource', 'quickstarts']
+        : [
+            `developer-${locale.locale}`,
+            `docs-${locale.locale}`,
+            `opensource-${locale.locale}`,
+            `quickstarts`,
+          ];
+      const results = await search({
```
Comment (sunnyzanchi) [inline src/pages/search-results.js:50]: we should handle the error case here. probably just an error message that says something like "unable to get search results, an error has occurred" is fine

File: src/pages/search-results.js [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+          font-weight: 500;
+        `}
+        dangerouslySetInnerHTML={{ __html: result.highlight.title }}
+      />
+      <p dangerouslySetInnerHTML={{ __html: result.highlight.body }} />
+    </Surface>
+  );
+};
+
+SearchResultPageView.propTypes = {
```
Comment (sunnyzanchi) [inline src/pages/search-results.js:256]: these proptypes don't match the component's implementation

File: src/pages/search-results.js [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+  Surface,
+  useLocale,
+  useQueryParams,
+} from '@newrelic/gatsby-theme-newrelic';
+
+import { usePagination, DOTS } from '../hooks/usePagination';
+
+const SearchResultPageView = ({ pageContext }) => {
+  const { queryParams } = useQueryParams();
+  const query = queryParams.get('query');
```
Comment (sunnyzanchi) [inline src/pages/search-results.js:20]: we should handle the case where `query` is `undefined` or an empty string. i think navigating to the home page would be fine

File: src/pages/search-results.js [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+      to={result.url}
+      css={css`
+        em {
+          color: #00ac69;
+          font-style: normal;
+        }
+        margin-bottom: 1rem;
+        box-shadow: none;
+        color: var(--primary-font-color);
+        &:hover {
```
Comment (sunnyzanchi) [inline src/pages/search-results.js:230]: after trying it out, i don't think hovering over a result that it's obvious enough you can click the entire result. what do you think about adding a `&:hover h3 { text-decoration: underline; }` here?

File: src/pages/search-results.js
Comment (sunnyzanchi) [inline src/pages/search-results.js:1]: the designs have a uniform left and right padding around the whole page. but bc of the media query in MainLayout, the right padding gets smaller at larger screen sizes. i didn't think it was that big of a deal but i noticed that this also means the pagination is off center so i think we should try and fix that

**Review by sunnyzanchi — CHANGES_REQUESTED** [review]
left a few comments but looks great overall! nice job!

File: src/pages/search-results.js (hunk 1 above)
Comment (LizBaker) [reply src/pages/search-results.js:50]: good point!

File: src/pages/search-results.js (hunk 4 above)
Comment (LizBaker) [reply src/pages/search-results.js:230]: poifect!

**Review by sunnyzanchi — APPROVED** [review]

**Pull request merged**